src/main.cpp

```diff
@@ -866,12 +919,36 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
 {
     // Basic checks that don't depend on any context
 
-    // Check transaction version
-    if (tx.nVersion < MIN_TX_VERSION) {
+    // Pre-Overwinter accepts version >=1 transaction. Overwinter only accepts version 3 transactions.
+    // To enable soft-forks of the transaction format, define a range of versions which can be accepted.
+    if (tx.fOverwintered) {
+        if (tx.nVersion < CTransaction::OVERWINTER_MIN_CURRENT_VERSION) {
+            return state.DoS(100, error("CheckTransaction(): overwinter version too low"),
+                REJECT_INVALID, "bad-tx-overwinter-version-too-low");
+        }
+        else if (tx.nVersion > CTransaction::OVERWINTER_MAX_CURRENT_VERSION ) {
+            return state.DoS(100, error("CheckTransaction(): overwinter version too high"),
+                REJECT_INVALID, "bad-tx-overwinter-version-too-high");
+        }
+    }
+
+    if (!tx.fOverwintered && tx.nVersion < MIN_TX_VERSION) {
```

Comment:
Add comment about how the previous consensus rule `tx.nVersion < MIN_TX_VERSION` is now split between here and `ContextualCheckTransaction`.

Reply:
Is it? I was writing a comment but then wasn't sure what you meant.  We'll have to add later after discussion.

Reply:
This ties into the other comments about the fact that the parser now ensures that `tx.nVersion` will never be negative, so this check which previously excluded negative values now is split between here (which excludes zero) and the `fOverwintered` check in `ContextualCheckTransaction` (which excludes what Sprout considers negative).